src/constants/label.js

```diff
@@ -130,42 +129,118 @@ export function localizeLayers(layers, locales) {
  * value. To reuse the evaluated value, bind it to a variable in a let
  * expression.
  *
- * @param haystack The overall string expression to search within.
- * @param needle The string to search for, or an expression that evaluates to
- *  this string.
+ * @param list The overall string expression to search within.
+ * @param separator A string to insert after the value, or an expression that
+ *  evaluates to this string.
+ * @param listStart A zero-based index into the list at which the search begins.
+ * @param numReplacements The maximum number of replacements remaining.
  */
-export function replaceExpression(
-  haystack,
-  needle,
-  replacement,
-  haystackStart,
-  numReplacements = 1
+function listValueExpression(
+  list,
+  separator,
+  valueToOmit,
+  listStart,
+  numReplacements
 ) {
-  let asIs = ["slice", haystack, haystackStart];
+  let asIs = ["slice", list, listStart];
   if (numReplacements <= 0) {
     return asIs;
   }
 
-  let needleStart = ["index-of", needle, haystack, haystackStart];
-  let needleLength =
-    typeof needle === "object" ? ["length", needle] : needle.length;
-  let needleEnd = ["+", needleStart, needleLength];
+  let iteration = numReplacements;
+  let rawSeparator = ";";
```

Comment:
I spilled a lot of ink in [this forum discussion](https://community.openstreetmap.org/t/multiple-delimited-names-in-the-name-tag/6803) about the need for a reliable delimiter&nbsp;– the semicolon. But what if we pretend that one of the popular ad-hoc delimiters is unambiguous enough to split on instead?

Sad reality | Simulation
----|----
[<img src="https://user-images.githubusercontent.com/1231218/211129038-b76003f3-dbb9-47b5-a2b2-f0aa98dc9cec.png" width="400" alt="Alicante before">](https://zelonewolf.github.io/openstreetmap-americana/#map=8.65/38.3721/-0.4708&language=en) | [<img src="https://user-images.githubusercontent.com/1231218/211129057-aff78816-2103-4b90-92b0-dbd68a75d25f.png" width="400" alt="Alicante after">](https://zelonewolf.github.io/openstreetmap-americana/#map=8.65/38.3721/-0.4708&language=en)
[<img src="https://user-images.githubusercontent.com/1231218/211129172-a9470a26-28fb-4f48-a626-188f4b9798b1.png" width="400" alt="Brussels before">](https://zelonewolf.github.io/openstreetmap-americana/#map=12/50.86018/4.34901&language=en) | [<img src="https://user-images.githubusercontent.com/1231218/211129188-79cff657-abb8-43cc-adc3-5f6f8a00239c.png" width="400" alt="Brussels after">](https://zelonewolf.github.io/openstreetmap-americana/#map=12/50.86018/4.34901&language=en)
[<img src="https://user-images.githubusercontent.com/1231218/211129267-cd693393-b7d7-4eba-9c73-2e4db0eab81c.png" width="400" alt="Casablanca before">](https://zelonewolf.github.io/openstreetmap-americana/#map=8/33.578/-7.378&language=en) | [<img src="https://user-images.githubusercontent.com/1231218/211129277-74b063cb-92b1-4a25-9684-93fbddb6dd81.png" width="400" alt="Casablanca after">](https://zelonewolf.github.io/openstreetmap-americana/#map=8/33.578/-7.378&language=en)
[<img src="https://user-images.githubusercontent.com/1231218/211129451-d8dfcd40-49d9-4a36-ba43-e026e91cbb61.png" width="400" alt="Hong Kong before">](https://zelonewolf.github.io/openstreetmap-americana/#map=10.5/22.3253/114.217&language=en) | [<img src="https://user-images.githubusercontent.com/1231218/211129467-0891403a-7acc-46de-9ba8-4769b992ac53.png" width="400" alt="Hong Kong after">](https://zelonewolf.github.io/openstreetmap-americana/#map=10.5/22.3253/114.217&language=en)

But alas, we can’t just turn every hyphen or slash or space into a value delimiter:

Sad reality | Sadder simulation
----|----
[<img src="https://user-images.githubusercontent.com/1231218/211129670-fd03c936-e1d4-4684-b0b4-f873bab49f66.png" width="400" alt="Tizi Ouzou and Bordj Bou Arreridjj before">](https://zelonewolf.github.io/openstreetmap-americana/#map=7/36.284/4.355&language=en) | [<img src="https://user-images.githubusercontent.com/1231218/211129702-3868c530-e785-43bb-a6f7-58a7b512b6f5.png" width="400" alt="Tizi Ouzou and Bordj Bou Arreridjj after">](https://zelonewolf.github.io/openstreetmap-americana/#map=7/36.284/4.355&language=en)
[<img src="https://user-images.githubusercontent.com/1231218/211129953-c640bd15-74eb-450b-8ca9-50913df23392.png" width="400" alt="Reynoldsburg - New Albany Road before">](https://zelonewolf.github.io/openstreetmap-americana/#map=14/40.06114/-82.8109/45&language=en) | [<img src="https://user-images.githubusercontent.com/1231218/211129973-77d31240-4e14-439d-9990-7ede44d63d7f.png" width="400" alt="Reynoldsburg - New Albany Road after">](https://zelonewolf.github.io/openstreetmap-americana/#map=14/40.06114/-82.8109/45&language=en)